src/mds/CDir.cc

```diff
@@ -3154,6 +3154,13 @@ void CDir::_freeze_dir()
 
   if (is_auth() && !is_subtree_root())
     inode->auth_pin(this);  // auth_pin for duration of freeze
+
+  if (get_inode()->is_export_ephemeral_distributed_migrating) {
+    auth = authority();
+    ceph_assert(auth.first >= 0);
+    ceph_assert(auth.second == CDIR_AUTH_UNKNOWN);
+    auth.second = auth.first;
+  }
```

Comment:
I don't think this change and most changes in Migrator.cc are not needed

Reply:
Ack dropped all of these. I had thought that only the child directory and not the entire child subtree needed to be exported. Fixed it.